taskvine/src/bindings/python3/ndcctools/taskvine/dask_dag.py

```diff
@@ -162,10 +234,14 @@ def set_result(self, key, value):
             else:
                 self._depth_of[r] = 0
 
-        for c in self._dependencies_of[key]:
-            self._pending_needed_by[c].discard(key)
+        if not DaskVineDag.taskref(self._working_graph[key]):
+            for c in self._dependencies_of[key]:
+                self._pending_needed_by[c].discard(key)
 
-        return rs.values()
+        if DaskVineDag.taskref(self._working_graph[key]):
+            return rs
```

Comment:
Is this mixing old and new graph styles?  If so, this file should only have the new style, and the one in `compat` the old style.